lib/Client.php

```diff
@@ -181,28 +210,146 @@ public function getFeatureFlag(
+            } catch (Exception $e) {
+                // TODO: handle error
+            }
+        }
+
+        return $response;
+
+    }
+
+    private function computeFlagLocally(
+        array $featureFlag,
+        string $distinctId,
+        array $groups = array(),
+        array $personProperties = array(),
+        array $groupProperties = array()
+    ) : bool | string
+    {
+        if ($featureFlag["ensure_experience_continuity"] ?? false) {
+            throw new InconclusiveMatchException("Flag has experience continuity enabled");
+        }
+
+        if (!$featureFlag["active"]) {
+            return false;
+        }
+
+        $flagFilters = $featureFlag["filters"] ?? [];
+        $aggregationGroupTypeIndex = $flagFilters["aggregation_group_type_index"] ?? null;
+
+        if (!is_null($aggregationGroupTypeIndex)) {
+            // TODO: handle groups
+            $groupName = $this->groupTypeMapping[strval($aggregationGroupTypeIndex)] ?? null;
+
+            if (is_null($groupName)) {
+                throw new InconclusiveMatchException("Flag has unknown group type index");
+            }
+
+            if (!array_key_exists($groupName, $groups)) {
+                return false;
+            }
+
+            $focusedGroupProperties = $groupProperties[$groupName];
+            return FeatureFlag::matchFeatureFlagProperties($featureFlag, $distinctId, $focusedGroupProperties);
+        } else {
+            return FeatureFlag::matchFeatureFlagProperties($featureFlag, $distinctId, $personProperties);
+        }
+
+    }
+
 
     /**
      * @param string $distinctId
      * @param array $groups
-     * @return array of enabled feature flags
+     * @return array of feature flags
      * @throws Exception
      */
-    public function fetchEnabledFeatureFlags(string $distinctId, array $groups = array()): array
+    public function fetchFeatureVariants(string $distinctId, array $groups = array(), array $personProperties = [], array $groupProperties = []): array
     {
-        $flags = json_decode($this->decide($distinctId, $groups), true)['featureFlags'] ?? [];
-        return array_filter($flags, function ($v) {
-            return $v != false;
-        });
+        $flags = json_decode($this->decide($distinctId, $groups, $personProperties, $groupProperties), true)['featureFlags'] ?? [];
+        return $flags;
     }
 
-    public function decide(string $distinctId, array $groups = array())
+    /**
+     * @throws Exception
+     */
+
+    private function loadFlags()
+    {
+        $payload = json_decode($this->localFlags(), true);
+        $this->featureFlags = $payload['flags'] ?? [];
+        $this->groupTypeMapping = $payload['group_type_mapping'] ?? [];
+    }
+
+
+    public function localFlags()
+    {
+        $payload = array(
+            'api_key' => $this->apiKey,
+        );
+
+        return $this->httpClient->sendRequest(
+            '/api/feature_flag/local_evaluation',
```

Comment:
Not quite sure about the vars here, I assume `api_key` is the project API key? All other libs pass it as the `token` param, so lets follow it? Reduces confusion.

```suggestion
            '/api/feature_flag/local_evaluation?token=<projectAPIkey>',
```

Also, we need to pass the `Authorization` header which has the personal API key. Haven't seen it yet 🤔 .

Also, some warning about local evaluation being disabled when personal API key is not provided.

Reply:
Ruby has everything in one PR, for reference. I had to do things a bit piecemeal for python as things kept coming up

Reply:
Try running this locally using an `example.php` file, should show issues like these, as our tests mock requests and thus can't find these :/